tools/ci_build/github/linux/docker/scripts/install_deps.sh

```diff
@@ -113,9 +113,7 @@ if [ $DEVICE_TYPE = "gpu" ]; then
     if [[ $BUILD_EXTR_PAR = *--enable_training* ]]; then
       ${PYTHON_EXE} -m pip install -r ${0/%install_deps.sh/training\/requirements.txt}
 
-      if [[ $BUILD_EXTR_PAR = *--enable_training_python_frontend_e2e_tests* || $BUILD_EXTR_PAR = *enable_training_pipeline_e2e_tests* ]]; then
-        source ${0/%install_deps.sh/install_openmpi.sh}
-      fi
+      source ${0/%install_deps.sh/install_openmpi.sh}
```

Comment:
Why are we removing that constraint?

Reply:
I need openmp for functions such as ```get_mpi_context_world_rank()``` and ```get_mpi_context_world_size()```. Otherwise, this errors out for me.

Reply:
Check with Liqun if we are allowe dto do that. Otherwise, we will install MPI in VMs that we shouldnt.

If you are getting this for the nightly pipeline, than something is wrong. If you are using a non-nightly, VM, why do we need MPI? MPI is for distributed training and "regular" VMs do not support it, so it doesnt make sense to install MPI on them

Reply:
you're right. It is not needed for non distributed scenario. Will remove it from the script since we are not going to run this test for the on commit verify job.